eos-payg-csv/eos-payg-csv

```diff
@@ -0,0 +1,180 @@
+            if header_norm != header_expected:
+                raise CsvParseError('''Provided CSV invalid:
+expected header:
+    {}
+got header:
+    {}'''.format(header_expected, row))
+
+            first_row = False
+        else:
+            expected_col_len = 5
+            col_len = len(row)
+            if col_len != expected_col_len:
+                raise CsvParseError('Provided CSV invalid: expected {} columns '
+                                    'but got {}'.format(expected_col_len, col_len))
+
+            key = row[-1]
+            expected_key_min = 64
+            key_len = len(key)
+            if key_len < expected_key_min:
+                raise CsvParseError('Provided CSV invalid: minimum key length '
+                                    '{} bytes but CSV contains a key of {} bytes'.format(
+                                        expected_key_min, key_len))
+
+        # we specifically want the last key for each device ID since there
+        # may be multiple entries for each device ID. Using a dict
+        # guarantees this.
+        id_to_key[row[0]] = row[-1]
+
+    # strip out an entry for the CSV header, if it exists
+    id_to_key.pop("device_id", None)
+
+    return id_to_key
+
+def write_time_codes(id_to_key):
+    files_written = []
+    for id, key in id_to_key.items():
+        key_filename = key_to_filename(id)
+        with open(key_filename, 'w') as f:
+            f.write(key)
+
+        codes_by_period = []
+        for period in PERIODS:
+            output = subprocess.run(['eos-payg-generate', key_filename, period],
+                                    stdout=subprocess.PIPE, check=True).stdout.decode('utf-8')
+            # prepend with a single quote to force Google Spreadsheets to treat
+            # every cell as a string, even if the default "convert text to
+            # numbers" option is chosen.
+            #
+            # This is needed to ensure leading zeros appear in the spreadsheet
+            # since users need to enter those as part of the given time code.
+            codes = ["'{}".format(code) for code in output.splitlines()]
+
+            codes_by_period.append(codes)
+
+        # rotate the matrix so instead of [1d, 1d, ...], [2d, 2d, ...], ...,
+        # we get: [1d, 2d, ...], ...
+        zipped = zip(*codes_by_period)
+
+        csv_filename = "{0}.csv".format(id)
+        with open(csv_filename, 'w', newline='') as csvfile:
+            csv_writer = csv.writer(csvfile)
+            headers = [period_to_display_str(period) for period in PERIODS]
+            csv_writer.writerow(headers)
+
+            for row in list(zipped):
+                csv_writer.writerow(row)
+        files_written.append(csv_filename)
+
+    return files_written
+
+def main(argv):
+    try:
+        id_to_key = acct_csv_get_id_to_key(argv[0])
+    except CsvParseError as cpe:
+        print("failed to read ACCT_KEY_CSV_FILE: {}".format(cpe),
```

Comment:
As above, I think this should be the actual filename rather than ACCT_KEY_CSV_FILE